### PR TITLE
squid:S1905 - Redundant casts should not be used

### PR DESCRIPTION
**Pull request by georgekankava**
This pull request is focused on resolving occurrences of Sonar rule squid:S1905 - Redundant casts should not be used.
You can find more information about the issue here:
https://dev.eclipse.org/sonar/coding_rules#q=squid%3AS1905
Please let me know if you have any questions.
George Kankava

### CODE REVIEW

**Comment by aaime** [comment]
Does not follow the contribution guide: https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md

File: modules/library/coverage/src/main/java/org/geotools/coverage/processing/operation/Crop.java [hunk 1]
```diff
@@ -445,7 +445,7 @@ public Coverage doOperation(ParameterValueGroup parameters, Hints hints) {
 					cropRoi, 
 					roiTolerance,
 					forceMosaic,
-                    (hints instanceof Hints) ? (Hints) hints: new Hints(hints),
+                    (hints instanceof Hints) ? hints: new Hints(hints),
```
Comment (jodygarnett) [inline modules/library/coverage/src/main/java/org/geotools/coverage/processing/operation/Crop.java]: API must of been changed, probably had the possibility of being a Map previously.

**Comment by jodygarnett** [comment]
While this change is too late for the 15-beta I encourage you to read the CONTIRBUTING.md guide (it should of been provided when you made the pull request?).

We would be happy to answer any questions you have.

**Comment by bencaradocdavies** [comment]
@georgekankava the code freeze is now over and you have an opportunity to get this on master.

**Comment by georgekankava** [comment]
@bencaradocdavies What should I do for this?

**Comment by bencaradocdavies** [comment]
@georgekankava updating the copyright headers is required to conform to the contribution guide linked above by @aaime: https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md

**Comment by georgekankava** [comment]
@bencaradocdavies Updated.

**Comment by bencaradocdavies** [comment]
Thanks @georgekankava.

**Pull request merged**